### PR TITLE
fix(java): display path parameters as variables in README

### PR DESCRIPTION
**Pull request by tstanmay13**
## Description
Linear ticket: Resolves [FER-6707: Snippets in the README don't properly show instantiating the client with variables and they show non-inlined path parameters like:](https://linear.app/buildwithfern/issue/FER-6707/snippets-in-the-readme-dont-properly-show-instantiating-the-client)

Path parameters in generated README examples now display as variable names (e.g., tenantId) instead of string literals (e.g., "tenant_id"), making the documentation more clear to actual usage. 

Changes made 
- See file changes 
- Handle edge cases like empty strings, special characters, and reserved keywords and fallback 

## Testing

- [x] Unit tests added/updated
- [x] Manual testing completed

### CODE REVIEW

File: generators/java-v2/dynamic-snippets/src/EndpointSnippetGenerator.ts [hunk 1]
```diff
@@ -710,9 +710,46 @@ export class EndpointSnippetGenerator {
             values: snippet.pathParameters ?? {}
         });
         for (const parameter of pathParameters) {
+            const paramValue = parameter.value;
+            let typeLiteral: java.TypeLiteral;
```
Comment (tjb9dc) [inline generators/java-v2/dynamic-snippets/src/EndpointSnippetGenerator.ts:714]: can we use `const` here? You might need to refactor this into a function that returns the typeLiteral

File: generators/java-v2/dynamic-snippets/src/EndpointSnippetGenerator.ts [hunk 2]
```diff
@@ -710,9 +710,46 @@ export class EndpointSnippetGenerator {
...
+
+            if (paramValue != null && typeof paramValue === "string") {
+                let variableName = String(paramValue)
+                    .replace(/[-_]+([a-z0-9])/gi, (_, char) => char.toUpperCase())
+                    .replace(/^([A-Z])/, (_, char) => char.toLowerCase())
+                    .replace(/[^a-zA-Z0-9]/g, "");
+
+                const JAVA_KEYWORDS = new Set([
```
Comment (tjb9dc) [inline generators/java-v2/dynamic-snippets/src/EndpointSnippetGenerator.ts:722]: I'm guessing this logic is duplicated here because the rest of the generator is written in Java?

**Review by tjb9dc — APPROVED** [review]

File: generators/java-v2/dynamic-snippets/src/EndpointSnippetGenerator.ts (hunk 1 above)
Comment (tstanmay13) [reply generators/java-v2/dynamic-snippets/src/EndpointSnippetGenerator.ts:714]: for sure

**Comment by tstanmay13** [comment]
Closing as this is not the right solution
